scipy/integrate/_ivp/ivp.py

```diff
@@ -0,0 +1,473 @@
+        Reason for algorithm termination:
+
+            * -1: Integration step failed.
+            * 0: The solver successfully reached the interval end.
+            * 1: A termination event occurred.
+
+    message : string
+        Verbal description of the termination reason.
+    success : bool
+        True if the solver reached the interval end or a termination event
+        occurred (``status >= 0``).
+
+    References
+    ----------
+    .. [1] J. R. Dormand, P. J. Prince, "A family of embedded Runge-Kutta
+           formulae", Journal of Computational and Applied Mathematics, Vol. 6,
+           No. 1, pp. 19-26, 1980.
+    .. [2] L. W. Shampine, "Some Practical Runge-Kutta Formulas", Mathematics
+           of Computation,, Vol. 46, No. 173, pp. 135-150, 1986.
+    .. [3] P. Bogacki, L.F. Shampine, "A 3(2) Pair of Runge-Kutta Formulas",
+           Appl. Math. Lett. Vol. 2, No. 4. pp. 321-325, 1989.
+    .. [4] E. Hairer, G. Wanner, "Solving Ordinary Differential Equations II:
+           Stiff and Differential-Algebraic Problems", Sec. IV.8.
+    .. [5] `Backward Differentiation Formula
+            <https://en.wikipedia.org/wiki/Backward_differentiation_formula>`_
+            on Wikipedia.
+    .. [6] L. F. Shampine, M. W. Reichelt, "THE MATLAB ODE SUITE", SIAM J. SCI.
+           COMPUTE., Vol. 18, No. 1, pp. 1-22, January 1997.
+    .. [7] `Stiff equation <https://en.wikipedia.org/wiki/Stiff_equation>`_ on
+           Wikipedia.
+    .. [8] A. Curtis, M. J. D. Powell, and J. Reid, "On the estimation of
+           sparse Jacobian matrices", Journal of the Institute of Mathematics
+           and its Applications, 13, pp. 117-120, 1974.
+    """
+    if method not in METHODS and not issubclass(method, OdeSolver):
+        raise ValueError("`method` must be one of {} or OdeSolver instance."
+                         .format(METHODS))
+
+    t0, tf = float(t_span[0]), float(t_span[1])
+
+    if t_eval is not None:
+        t_eval = np.asarray(t_eval)
+        if t_eval.ndim != 1:
+            raise ValueError("`t_eval` must be 1-dimensional.")
+
+        if np.any(t_eval < min(t0, tf)) or np.any(t_eval > max(t0, tf)):
+            raise ValueError("Values in `t_eval` are not within `t_span`")
+
+        d = np.diff(t_eval)
+        if tf > t0 and np.any(d <= 0) or tf < t0 and np.any(d >= 0):
+            raise ValueError("Values in `t_eval` are not properly sorted.")
+
+        if tf > t0:
+            t_eval_i = 0
+        else:
+            # Make order of t_eval ascending to use np.searchsorted.
+            t_eval = t_eval[::-1]
+            # This will be an upper bound for slices.
+            t_eval_i = t_eval.shape[0]
+
+    if not callable(method):
+        method = METHODS[method]
+
+    solver = method(fun, t0, y0, tf, vectorized=vectorized, **options)
+
+    if t_eval is None:
+        ts = [t0]
+        ys = [y0]
+    else:
+        ts = []
+        ys = []
+
+        # This code needs clarification.
```

Comment:
If there is only one `t_eval` and it is equal to `t0`, then this code prevents the solver from taking an unnecessary step. It also ensures that when `t_eval[0] == t0` then `ys[0] == y0`.  I added this when I couldn't figure out why the solver was stepping even when it didn't need to and why sometimes my first point was slightly different from the initial condition. (I have some weird problems.) It is not strictly necessary.

If you decide to keep it, the unnecessary `pass` should be removed. I can add a test for the behavior described above.